airbyte-integrations/connectors/source-stock-ticker-api/source_stock_ticker_api/source.py

```diff
@@ -0,0 +1,107 @@
+
+logger = logging.getLogger("airbyte")
+
+
+class ExchangeRates(HttpStream, IncrementalMixin):
+    url_base = "http://api.exchangeratesapi.io/"
+    cursor_field = "date"
+    primary_key = "date"
+
+    def __init__(self, config: Mapping[str, Any], start_date: datetime, **kwargs):
+        super().__init__()
+        self.base = config['base']
+        self.access_key = config['access_key']
+        self.start_date = start_date
+        self._cursor_value = None
+
+    def path(
+            self,
+            stream_state: Mapping[str, Any] = None,
+            stream_slice: Mapping[str, Any] = None,
+            next_page_token: Mapping[str, Any] = None) -> str:
+        return stream_slice['date']
+
+    def request_params(
+            self,
+            stream_state: Mapping[str, Any],
+            stream_slice: Mapping[str, Any] = None,
+            next_page_token: Mapping[str, Any] = None,
+    ) -> MutableMapping[str, Any]:
+        # The api requires that we include access_key as a query param so we do that in this method
+        return {'access_key': self.access_key}
+
+    def read_records(self, *args, **kwargs) -> Iterable[Mapping[str, Any]]:
+        for record in super().read_records(*args, **kwargs):
+            if self._cursor_value:
+                latest_record_date = datetime.strptime(record[self.cursor_field], '%Y-%m-%d')
+                self._cursor_value = max(self._cursor_value, latest_record_date)
+            yield record
+
+    def parse_response(
+            self,
+            response: requests.Response,
+            stream_state: Mapping[str, Any],
+            stream_slice: Mapping[str, Any] = None,
+            next_page_token: Mapping[str, Any] = None,
+    ) -> Iterable[Mapping]:
+        # The response is a simple JSON whose schema matches our stream's schema exactly,
+        # so we just return a list containing the response
+        return [response.json()]
+
+    def next_page_token(self, response: requests.Response) -> Optional[Mapping[str, Any]]:
+        # The API does not offer pagination,
+        # so we return None to indicate there are no more pages in the response
+        return None
+
+    def _chunk_date_range(self, start_date: datetime) -> List[Mapping[str, Any]]:
+        """
+        Returns a list of each day between the start date and now.
+        The return value is a list of dicts {'date': date_string}.
+        """
+        dates = []
+        while start_date < datetime.now():
+            dates.append({self.cursor_field: start_date.strftime('%Y-%m-%d')})
+            start_date += timedelta(days=1)
+        return dates
+
+    def stream_slices(self, sync_mode, cursor_field: List[str] = None, stream_state: Mapping[str, Any] = None) -> Iterable[Optional[Mapping[str, Any]]]:
+        start_date = datetime.strptime(stream_state[self.cursor_field], '%Y-%m-%d') if stream_state and self.cursor_field in stream_state else self.start_date
```

Comment:
`datetime.strptime(*, '%Y-%m-%d')` is reused heavily, please move it to a separate function